docs/implementation_design/packages_plugins.md

```diff
@@ -1 +1,14 @@
 # Packages and plugins
```

Comment:
Is this doc about the high level overview of how packages and plugins work together? if so maybe change the title to something more descriptive like "Dependency of packages and plugins"

Reply:
Good point! ;)